sycl/include/sycl/stl_wrappers/assert.h

```diff
@@ -0,0 +1,44 @@
+//== ----------------<assert.h> wrapper around STL--------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// Must not be guarded. C++ standard says the macro assert is redefined
+// according to the current state of NDEBUG each time that <cassert> is
+// included.
+
+#if defined(__has_include_next)
+#include_next <assert.h>
+#else
+#include <../ucrt/assert.h>
+#endif
+
+#ifdef __SYCL_DEVICE_ONLY__
+#include <CL/__spirv/spirv_vars.hpp>
```

Comment:
@lbushi25, I know this PR has already been merged, but... Is there any reason this inclusion can't be moved inside the `_WIN32` conditional block below? That would avoid possible interference on non-Windows systems. Additionally, this inclusion only appears to be needed when `NDEBUG` is not defined; including it only when needed might help to (slightly) reduce compilation overhead.

Same concerns for the `<cassert>` wrapper.

Reply:
Hmm, I agree. I'll have a look.